aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -465,19 +447,41 @@ impl AptosVM {
         }
     }
 
+    fn inject_abort_info_if_available(&self, status: ExecutionStatus) -> ExecutionStatus {
+        match status {
+            ExecutionStatus::MoveAbort {
+                location: AbortLocation::Module(module),
+                code,
+                ..
+            } => {
+                let info = self
+                    .extract_module_metadata(&module)
+                    .and_then(|m| m.extract_abort_info(code));
+                ExecutionStatus::MoveAbort {
+                    location: AbortLocation::Module(module),
+                    code,
+                    info,
+                }
+            },
+            _ => status,
+        }
+    }
+
     fn finish_aborted_transaction(
         &self,
         gas_meter: &mut impl AptosGasMeter,
         txn_data: &TransactionMetadata,
         resolver: &impl AptosMoveResolver,
+        status: ExecutionStatus,
         log_context: &AdapterLogSchema,
         change_set_configs: &ChangeSetConfigs,
-    ) -> Result<(VMChangeSet, FeeStatement), VMStatus> {
+    ) -> Result<(VMChangeSet, FeeStatement, ExecutionStatus), VMStatus> {
         // Storage refund is zero since no slots are deleted in aborted transactions.
         const ZERO_STORAGE_REFUND: u64 = 0;
 
         if is_account_init_for_sponsored_transaction(txn_data, &self.features) {
             let mut session = self.new_session(resolver, SessionId::run_on_abort(txn_data));
+            let status = self.inject_abort_info_if_available(status);
```

Comment:
Could this be moved to the top of the `if .. else` to reduce code duplication?

Reply:
so the fun thing here is that apparently calling "new_session" resets some internal state of the VM, which in turn causes the modules loaded to be dropped and hence the abort info is cleared.

Reply:
Document that? too easy a clean up that I can do any day if I tough this part of code.